rust/arrow/src/datatypes.rs

```diff
@@ -151,19 +152,70 @@ make_type!(UInt64Type, u64, DataType::UInt64, 64, 0u64);
 make_type!(Float32Type, f32, DataType::Float32, 32, 0.0f32);
 make_type!(Float64Type, f64, DataType::Float64, 64, 0.0f64);
 
-/// A subtype of primitive type that represents numeric values.
-pub trait ArrowNumericType: ArrowPrimitiveType {}
-
-impl ArrowNumericType for Int8Type {}
-impl ArrowNumericType for Int16Type {}
-impl ArrowNumericType for Int32Type {}
-impl ArrowNumericType for Int64Type {}
-impl ArrowNumericType for UInt8Type {}
-impl ArrowNumericType for UInt16Type {}
-impl ArrowNumericType for UInt32Type {}
-impl ArrowNumericType for UInt64Type {}
-impl ArrowNumericType for Float32Type {}
-impl ArrowNumericType for Float64Type {}
+/// A subtype of primitive type that represents numeric values, if available a SIMD type and
+/// SIMD operations are defined in this trait.  SIMD is leveraged in the `compute` module if
+/// available.
+pub trait ArrowNumericType: ArrowPrimitiveType {
+    /// Defines the SIMD type that should be used for this numeric type
+    #[cfg(any(target_arch = "x86", target_arch = "x86_64"))]
```

Comment:
Can we put this above the trait definition? also, I'm not sure if we should define another trait just for SIMD, since now `ArrowNumericType` really is almost all about SIMD.

Reply:
It looked wrong to have `#[cfg(any(target_arch = "x86", target_arch = "x86_64"))]` in multiple places, I plan to go back and see how to clean this up.

> also, I'm not sure if we should define another trait just for SIMD, since now ArrowNumericType really is almost all about SIMD.

I don't quite understand what you mean here?  What do you propose?

Reply:
Right, that probably does make sense as eventually we will need SIMD ops over Boolean Arrays as well.